### DISCUSSION TITLE
TTS and Media Overlays: granular skipability and escapability user preferences

### DISCUSSION THREAD

**Discussion by oscarlcarlsson** [edrlab/thorium-reader#2655] [Ideas]
Would it be possible to add a more granular filter to the skipability funktion?
In our current titles we have several elements that our users might skip one or more of: i.e. a user might want to skip page numbers but not image descriptions.

In our latest discussion at MTM, we have found the following elements where it would be good to skip some and not others.
	- [ ] pagenumbers
	- [ ] image/table descriptions
	- [ ] footnotes/endnotes
	- [ ] aside
	- [ ] alt-text

A general skip-all-button is still useful. But a more detailed list would be helpful.

**Comment by oscarlcarlsson** [comment]
For the smil-files, this would consider the following epub:type-attributes:
	- [ ] epub:type="pagebreak"
	- [ ] epub:type="aside"
	- [ ] epub:type="noteref"
	- [ ] epub:type="footnote"

**Comment by danielweck** [comment]
I have converted your original issue to a discussion so that we can come up with a UX design that doesn't introduce too much complexity in the GUI. We need to draw up a list of individually-selectable "skippable" semantic roles that can be mapped to TTS readalaoud and Media Overlays playback. Whilst we are at it, I suggest we look into "escapable" semantics too (currently only implemented for TTS, behind a keyboard shortcut modifier for previous/next navigation commands (as well as the key mod when clicking on the respective icon buttons in the top toolbar)

**Comment by danielweck** [comment]
ESCAPE:
* TTS: https://github.com/readium/r2-navigator-js/blob/822d834d5e9159c083927d96c583e08a527704c6/src/electron/renderer/webview/readaloud.ts#L410-L426
* MO: https://github.com/readium/r2-navigator-js/blob/822d834d5e9159c083927d96c583e08a527704c6/src/electron/renderer/media-overlays.ts#L801-L857

SKIP:

* TTS: https://github.com/readium/r2-navigator-js/blob/822d834d5e9159c083927d96c583e08a527704c6/src/electron/renderer/common/dom-text-utils.ts#L302-L341
* MO: https://github.com/readium/r2-navigator-js/blob/822d834d5e9159c083927d96c583e08a527704c6/src/electron/renderer/media-overlays.ts#L780-L799

**Reply by oscarlcarlsson** [reply]
When I had a look at the elements in https://www.w3.org/TR/epub-ssv-11/ i noticed that several of them has been replaced by `<aside>`.
Shoudn't `<aside>` be included in the list in SKIP-MO?

**Reply by danielweck** [reply]
Well, the EPUB specifications are quite permissive (i.e. not prescriptive through strong SHOULD/MUST normative statements) when it comes to defining conformance requirements for reading systems related to skippability / escapability. We (as a community) will need to agree on a baseline for skippables / escapables in real-world usage, in order to harmonise implementation efforts in the ecosystem. See https://github.com/daisy/epub-accessibility-tests/issues/122

**Comment by danielweck** [comment]
Related discussion: https://github.com/edrlab/thorium-reader/discussions/2679

**Reply by danielweck** [reply]
Related:
* https://github.com/readium/guided-navigation/discussions/3
* https://github.com/w3c/publishingcg/issues/72

**Reply by danielweck** [reply]
Resource:
* https://www.w3.org/TR/epub-a11y-tech-11/#sync-skippability
* https://www.w3.org/TR/epub-a11y-tech-11/#sync-escapability
